opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -137,10 +137,32 @@ mod tests {
             }
         }
 
-        fn on_end(&self, _span: SpanData) {
-            // TODO: Accessing Context::current() will panic today and hence commented out.
+        fn on_end(&self, span: SpanData) {
+            // Fixed: Context::current() no longer panics from Drop
             // See https://github.com/open-telemetry/opentelemetry-rust/issues/2871
-            // let _c = Context::current();
+            let current_cx = Context::current();
+
+            // IMPORTANT NOTE: The context returned here is the PARENT context,
```

Comment:
This comment is still a bit missleading.

Reply:
Thanks. Fixed 
https://github.com/open-telemetry/opentelemetry-rust/pull/3262/commits/6c6a35b12cd8a840c121679842a0c3cfba578af5